src/overlays/actors/ovl_Bg_Spot01_Objects2/z_bg_spot01_objects2.c

```diff
@@ -31,17 +29,113 @@ const ActorInit Bg_Spot01_Objects2_InitVars = {
     (ActorFunc)BgSpot01Objects2_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Objects2/BgSpot01Objects2_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Objects2/BgSpot01Objects2_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 12800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+Gfx* D_808AC510[] = { 0x06001EB0, 0x06002780, 0x06003078, 0x06001228, 0x06001528 };
+
+extern UNK_TYPE D_06001A38;
+extern UNK_TYPE D_06001C58;
+
+void BgSpot01Objects2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot01Objects2* this = THIS;
+
+    switch (this->dyna.actor.params & 7) {
+        case 0:
+        case 1:
+        case 2:
+            this->objectId = OBJECT_SPOT01_MATOYA;
+            break;
+        case 3:
+            this->objectId = OBJECT_SPOT01_MATOYAB;
+            break;
+        case 4:
+            this->objectId = OBJECT_SPOT01_MATOYA;
+    }
+
+    if (this->objectId >= 0) {
+        this->objBankIndex = Object_GetIndex(&globalCtx->objectCtx, this->objectId);
+        if (this->objBankIndex < 0) {
+            // There was no bank setting.
+            osSyncPrintf("-----------------------------バンク設定ありませんでした.");
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+    } else {
+        Actor_Kill(&this->dyna.actor);
+    }
+    this->actionFunc = func_808AC2BC;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+}
+
+void BgSpot01Objects2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 func_808AC22C(Path* pathList, Vec3f* pos, s32 path, s32 waypoint) {
+    Vec3s* pointPos = &((Vec3s*)SEGMENTED_TO_VIRTUAL(((Path*)pathList + path)->points))[waypoint];
```

Comment:
```suggestion
    Vec3s* pointPos = &((Vec3s*)SEGMENTED_TO_VIRTUAL(pathList[path].points))[waypoint];
```

Reply:
Unfortunately doing it like this causes regalloc issues and a non matching function 

```
    Vec3s* pointPos = &((Vec3s*)SEGMENTED_TO_VIRTUAL((pathList + path)->points))[waypoint];

```
I've removed the cast to Path * and this matches, but accessing the path with [] breaks the function... pretty weird....

Reply:
Hmm, not very surprising but it's odd. We can go with it for now but it will probably be improved somehow later.